src/v/cloud_topics/level_zero/gc/level_zero_gc.h

```diff
@@ -179,7 +179,7 @@ class level_zero_gc {
     /*
      * Interface for computing the maximum epoch eligible for GC.
      */
-    class epoch_source {
+    class cluster_support {
```

Comment:
I liked epoch_source because it was at least clear that it derived its outputs from epochs (and if we change last_applied to be cluster_epoch, all of its functions would be returning epochs, which makes "source" feel like an apt name). IMO cluster_support doesn't tell users much about what they should expect from this class.